doc/modules/rbl.md

```diff
@@ -137,3 +143,244 @@ rbls {
+	# Check Reply-to header
+	replyto = true;
+	# Check only domain part
+        domain_only = true;
+      }
+  }
+}
+~~~
+
+## Principles of operation
+
+In this section, we define how `surbl` module performs its checks.
+
+### TLD composition
+
+By default, we want to check some top level domain, however, many domains contain
+two components while others can have 3 or even more components to check against the
+list. By default, rspamd takes top level domain as defined in the [public suffixes](https://publicsuffix.org).
+Then one more component is prepended, for example:
+
+    sub.example.com -> [.com] -> example.com
+    sub.co.uk -> [.co.uk] -> sub.co.uk
+
+However, sometimes even more levels of domain components are required. In this case,
+the `exceptions` map can be used. For example, if we want to check all subdomains of
+`example.com` and `example.co.uk`, then we can define the following list:
+
+    example.com
+    example.co.uk
+
+Here are new composition rules:
+
+    sub.example.com -> [.example.com] -> sub.example.com
+    sub1.sub2.example.co.uk -> [.example.co.uk] -> sub2.example.co.uk
+
+### DNS composition
+
+SURBL module composes the DNS request of two parts:
+
+- TLD component as defined in the previous section;
+- DNS list suffix
+
+For example, to form a request to multi.surbl.org, the following applied:
+
+    example.com -> example.com.multi.surbl.com
+
+### Results parsing
+
+Normally, DNS blacklists encode reply in A record from some private network
+(namely, `127.0.0.0/8`). Encoding varies from one service to another. Some lists
+use bits encoding, where a single DNS list or error message is encoded as a bit
+in the least significant octet of the IP address. For example, if bit 1 encodes `LISTA`
+and bit 2 encodes `LISTB`, then we need to perform bitwise `OR` for each specific bit
+to decode reply:
+
+     127.0.0.3 -> LISTA | LISTB -> both bit symbols are added
+     127.0.0.2 -> LISTB only
+     127.0.0.1 -> LISTA only
+
+This encoding can save DNS requests to query multiple lists one at a time.
+
+Some other lists use direct encoding of lists by some specific addresses. In this
+case you should define results decoding principle in `ips` section not `bits` since
+bitwise rules are not applicable to these lists. In `ips` section you explicitly
+match the ip returned by a list and its meaning.
+
+## IP lists
+
+From rspamd 1.1 it is also possible to do two step checks:
+
+1. Resolve IP addresses of each URL
+2. Check each IP resolved against SURBL list
+
+In general this procedure could be represented as following:
+
+* Check `A` or `AAAA` records for `example.com`
+* For each ip address resolve it using reverse octets composition: so if IP address of `example.com` is `1.2.3.4`, then checks would be for `4.3.2.1.uribl.tld`
+
+For example, [SBL list](https://www.spamhaus.org/sbl/) of `spamhaus` project provides such functions using `ZEN` multi list. This is included in rspamd default configuration:
```

Comment:
```suggestion
For example, [SBL list](https://www.spamhaus.org/sbl/) of `Spamhaus` project provides such functions using `ZEN` multi list. This is included in rspamd default configuration:
```

Reply:
SBL was disabled by [default config](https://github.com/rspamd/rspamd/blob/e16042d0d4db18d29724f596e735dde321675ed5/conf/modules.d/rbl.conf#L274) and I suggest to exclude information about SBL.

Reply:
Agree, `resolve_ip` is evil.